src/helpers/credential.rs

```diff
@@ -269,6 +279,138 @@ impl<'a> Credential<'a> {
         Ok(())
     }
 
+    pub async fn create_self_issued_credential(
+        &mut self,
+        schema_did: &str,
+        credential_subject_str: &str,
+        bbs_secret: &str,
+        bbs_private_key: &str,
+        credential_revocation_did: Option<&str>,
+        credential_revocation_id: Option<&str>,
+        exp_date: Option<&str>,
+    ) -> Result<String, CredentialError> {
+        let credential_subject: CredentialSubject = serde_json::from_str(credential_subject_str)?;
+        let credential_subject_copy: CredentialSubject = serde_json::from_str(credential_subject_str)?;
+        let subject_did = credential_subject.id.ok_or("no subject did found".to_string()).unwrap();
+        let subject_did_copy = subject_did.clone();
+        let subject_did_str = subject_did_copy.as_str();
+        let subject_did_str_copy = subject_did_str.clone();
```

Comment:
Should we use 4 variables for the same thing? :D Could you just use `.clone()` where needed directly? `subject_did.clone()`

Reply:
Done :)